src/algebraic_geometry/prime_spectrum.lean

```diff
@@ -69,6 +72,103 @@ def zero_locus (s : set R) : set (prime_spectrum R) :=
 @[simp] lemma mem_zero_locus (x : prime_spectrum R) (s : set R) :
   x ∈ zero_locus s ↔ s ⊆ x.as_ideal := iff.rfl
 
+@[simp] lemma zero_locus_span (s : set R) :
+  zero_locus (ideal.span s : set R) = zero_locus s :=
+by { ext x, exact (submodule.gi R R).gc s x.as_ideal }
+
+/-- The vanishing ideal of a set `s` of points
+of the prime spectrum of a commutative ring `R`
+is the largest ideal of the ring that is contained in all the prime ideals in the set `s`.
+
+An element `f` of `R` can be thought of as a dependent function
+on the prime spectrum of `R`.
+At a point `x` (a prime ideal)
+the function (i.e., element) `f` takes values in the quotient ring `R` modulo the prime ideal `x`.
+In this manner, `vanishing_ideal s` is exactly the ideal of `R`
+consisting of all "functions" that vanish on all of `s`.
+-/
+def vanishing_ideal (s : set (prime_spectrum R)) : ideal R :=
```

Comment:
I think it would help readers a lot if you could have some kind of naming convention in this file, where `s` would always be a set in `R`, and `t` would always be a set in `prime_spectrum R`.

Reply:
Doen